### ISSUE TITLE
"with" syntax and Dictionaries

### ISSUE THREAD

**Issue by flucas1** [meziantou/Meziantou.Analyzer#1235]
### Version of the Meziantou.Analyzer NuGet package

3.0.129

### Rule Identifier

MA0002

### Target Framework

net11.0

### C# Language version

preview

### Description

I expect this to be valid code for the analyzer

`Dictionary<string, List<string>> duplicates = [with(StringComparer.Ordinal)];`

But it raises MA0002

### Reproduction Steps

Minimal code:

```c#
Dictionary<string, List<string>> duplicates = [with(StringComparer.Ordinal)];
```

**Comment by flucas1** [comment]
As an additonal comment the with syntax also applies to ConcurrentDictionary<string,..>, HashSet<string> ImmutableHashSet<string> and none of these cases is detected.

I would expect this to be OK (as it is)
`ImmutableHashSet<string> references = [with(StringComparer.Ordinal), .. otherHashset];`

But this to raise a warning
`ImmutableHashSet<string> references = [.. otherHashset];`
